iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/function_ref.inl

```diff
@@ -34,12 +49,17 @@ template <class ReturnType, class... ArgTypes>
 inline function_ref<ReturnType(ArgTypes...)>::function_ref(ReturnType (&function)(ArgTypes...)) noexcept
 {
     // the cast is required to work on POSIX systems
+    // AXIVION Next Line AutosarC++19_03-A5.2.4, AutosarC++19_03-A5.2.4-M5.2.6 : Type-safety ensured by casting back on
+    // call
     m_pointerToCallable = reinterpret_cast<void*>(function);
 
     // the lambda does not capture and is thus convertible to a function pointer
     // (required by the C++ standard)
     m_functionPointer = [](void* target, ArgTypes... args) -> ReturnType {
-        auto f = reinterpret_cast<ReturnType (*)(ArgTypes...)>(target);
+        using PointerType = ReturnType (*)(ArgTypes...);
+        // AXIVION Next Line AutosarC++19_03-A5.2.4 : Type-safety ensured by casting from type
```

Comment:
@saif-at-github Axivion is having problems to detect the suppression of `AutosarC++19_03-A5.2.4`. I still get 4 hits when analyzing this branch? Do you know why this is happening? @dkroenke had the same issue.